typing/typecore.ml

```diff
@@ -778,21 +791,66 @@ let solve_constructor_annotation
         | _ -> assert false
   in
   if ids <> [] then ignore begin
```

Comment:
I wonder why this `ignore` is here and if it is still necessary.

Reply:
Indeed it was spurious; removed it.